src/bin/enclave-runc-wrapper.rs

```diff
@@ -0,0 +1,241 @@
+enum NewContainerError {
+    #[error("could not retrieve the runc status")]
+    Status(#[from] std::io::Error),
+
+    #[error("could not format")]
+    Format(#[from] std::fmt::Error),
+
+    #[error("could not convert bytes to utf-8 string")]
+    Utf8(#[from] std::string::FromUtf8Error),
+
+    #[error("could not parse JSON")]
+    Json(#[from] serde_json::Error),
+}
+
+fn container_namespace(
+    container_id: &str,
+    container_root: Option<std::string::String>,
+) -> Result<Option<std::string::String>, NewContainerError> {
+    let mut state_cmd = std::process::Command::new("runc");
+    if let Some(ref v) = container_root {
+        state_cmd.arg("--root").arg(v);
+    }
+
+    state_cmd.arg("state").arg(container_id);
+    let state_raw = String::from_utf8(state_cmd.output()?.stdout)?;
+
+    let state: serde_json::Value = serde_json::from_str(&state_raw)?;
+    let annotations_o = state["annotations"].as_object();
+
+    match annotations_o {
+        Some(annotations) => {
+            // containerd
+            if annotations.contains_key(ANNOTATION_CONTAINERD_LOG_DIRECTORY) {
+                // containerd doesn't expose k8s namespaces directly. They have
+                // to be parsed from the log directory path, where the first
+                // part of the filename is the namespace.
+                let log_directory = annotations[ANNOTATION_CONTAINERD_LOG_DIRECTORY]
+                    .as_str()
+                    .unwrap();
+                let p = std::path::PathBuf::from(log_directory);
+                let file_name = p.file_name().unwrap().to_str().unwrap();
+                let mut splitter = file_name.split('_');
+                let namespace = splitter.next().unwrap().to_string();
+
+                return Ok(Some(namespace));
+            } else if annotations.contains_key(ANNOTATION_CONTAINERD_SANDBOX_ID) {
+                // When a container is running as a part of a previously created
+                // pod, the log directory path has to be retrieved from the
+                // sandbox container.
+                let sandbox_id = annotations[ANNOTATION_CONTAINERD_SANDBOX_ID]
+                    .as_str()
+                    .unwrap();
+                return container_namespace(sandbox_id, container_root);
+            }
+            Ok(None)
+        }
+        None => Ok(None),
+    }
+}
+
+async fn policy_label(
+    namespace_o: Option<std::string::String>,
+) -> Result<enclave::ContainerPolicyLevel, kube::Error> {
+    let namespace_s: std::string::String;
+
+    match namespace_o {
+        Some(v) => namespace_s = v,
+        None => return Ok(enclave::ContainerPolicyLevel::Baseline),
+    }
+
+    // Apply the privileged policy for kube-system containers immediately.
+    // Otherwise the core k8s components (apiserver, scheduler) won't be able
+    // to run.
+    if namespace_s.as_str() == "kube-system" {
+        return Ok(enclave::ContainerPolicyLevel::Privileged);
+    }
```

Comment:
This may be able to re-written as
```rust
let namespace_s = match namespace_o {
  Some(v) if v.as_str() == "kube-system" => return Ok(enclave::ContainerPolicyLevel::Privileged),
  Some(v) => v,
  None => return Ok(enclave::ContainerPolicyLevel::Baseline),
}
```
There's probably a more elegant way to do this though that's not coming to mind. Apologies for that!